### PR TITLE
Default config in plugin Code

### PR DESCRIPTION
**Pull request by xbill82**
Adapts the plugin to the Apache-ish installation system specified in kuzzleio/kuzzle#609

The plugin has now its own default config in its own code and can run with a null customConfig passed to the init function.

## Boyscouting
Bumped the version of `rewired` and `should`. Added `sinon` as a devDependency.
Got rid of Grunt.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,31 +23,24 @@
...
-      "strategies": {}
-    }
-  },
   "license": "Apache-2.0",
   "dependencies": {
     "passport": "0.3.2",
     "passport-oauth2": "1.x.x",
     "passport-oauth2-refresh": "1.0.0",
-    "q": "2.0.3"
+    "q": "2.0.3",
```
Comment (scottinet) [inline package.json]: // TODO: replace `q` with `bluebird`

I know this is not exactly the right PR to do that, but do you think you can apply this change?
`q` is really slow and memory consuming compared to other promise modules.

File: package.json [hunk 2]
```diff
@@ -23,31 +23,24 @@
...
+    "sinon": "^1.17.7"
```
Comment (scottinet) [inline package.json]: `sinon` should be listed as a `devDependencies` module

File: package.json [hunk 3]
```diff
@@ -23,31 +23,24 @@
...
-      "strategies": {}
-    }
-  },
   "license": "Apache-2.0",
   "dependencies": {
     "passport": "0.3.2",
     "passport-oauth2": "1.x.x",
     "passport-oauth2-refresh": "1.0.0",
-    "q": "2.0.3"
+    "bluebird": "2.0.3"
```
Comment (ballinette) [inline package.json]: `bluebird` is not needed. At most it is a devDepencency, but you can also remove it entirely (see below).

File: test/verify.test.js [hunk 4]
```diff
@@ -77,7 +77,7 @@ describe('passport verify', function() {
       get: function () {
         return {
           load: function () {
-            return q(null);
+            return new q(() => {});
```
Comment (ballinette) [inline test/verify.test.js]: can be replaced by `return Promise.resolve({})` to remove the dependency with `bluebird`

**Pull request merged**